Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift

```diff
@@ -214,4 +226,76 @@ class AssignmentDetailsPresenter {
             break
         }
     }
+
+    private func showOnlineUpload() {
+        let filePicker = FilePickerViewController.create()
+        filePicker.title = NSLocalizedString("Submission", bundle: .student, comment: "")
+        filePicker.cancelButtonTitle = NSLocalizedString("Cancel Submission", bundle: .student, comment: "")
+        filePicker.delegate = self
+        let allowedUTIs = assignment?.allowedUTIs ?? []
+        if allowedUTIs.contains(where: { $0.isAny || $0.isImage || $0.isVideo }) {
+            filePicker.sources = [.camera, .library, .files]
+        } else {
+            filePicker.sources = [.files]
+        }
+        filePicker.files = self.files.map { $0 }
+        filePicker.utis = allowedUTIs
+        self.filePicker = filePicker
+        let nav = UINavigationController(rootViewController: filePicker)
+        view?.present(nav, animated: true, completion: nil)
+    }
+}
+
+extension AssignmentDetailsPresenter: FilePickerControllerDelegate {
```

Comment:
Yeah, I think this delegate pattern will work a little better than having the `FilePickerPresenterProtocol`